File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.py
class Solution(object):
    def maxDistance(self, colors):
        """
        :type colors: List[int]
        :rtype: int
        """

        result = 0 
        for i in range ( 0 , len(colors) ):

            index = len(colors) - 1

            while( index > i and colors[i] == colors[index]):
                index -= 1 

            result = max( result , abs( i - index ))

        return result 
        

File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.cpp
class Solution {
public:
    int maxDistance(vector<int>& colors) {
        
        int result = 0 ;

        for( int i = 0 ; i < colors.size() ; i++ ){

            int index = colors.size() - 1 ;
            while( index > i && colors[i] == colors[index] ) index -- ;

            result = max( result , abs( i - index ));
        }

        return result ;
    }
};